salt/etc-hosts/init.sls

```diff
@@ -1,20 +1,19 @@
 {# In Kubernetes, /etc/hosts is mounted in from the host. file.blockreplace fails on this #}
 {% if 'ca' not in salt['grains.get']('roles', []) %}
 /etc/hosts:
-  file.blockreplace:
-    # If markers are changed, also update etc-hosts/update-pre-reboot.sls
+  caasp_hosts.managed:
     - marker_start: "#-- start Salt-CaaSP managed hosts - DO NOT MODIFY --"
     - marker_end:   "#-- end Salt-CaaSP managed hosts --"
-    - source:       salt://etc-hosts/hosts.jinja
-    - template:     jinja
-    - append_if_not_found: True
 {% else %}
 {# See https://github.com/saltstack/salt/issues/14553 #}
 etc-hosts-dummy:
   cmd.run:
     - name: "echo saltstack bug 14553"
 {% endif %}
 
+# TODO: all these `docker cp` could be removed:
+#       we could just mount the /etc/hosts file
```

Comment:
Should we do this in a follow-up PR?

Reply:
docker treats /etc/hosts as special, it can't be mounted in.

Reply:
Hmmm, I am pretty sure you can mount `/etc/hosts`? Our problem before was that it was an atomic write if it was modified, and once you mounted it you weren't able to update it, so we had to `docker cp` it.

Reply:
Yeah, I'm not sure you can't mount either. I think the real problem was that the bind-mount really references an inode, so Salt was updating the file by creating a new copy of the file and overwritting the old copy, while the container kept the old copy... In contrast, this Python code does the replacement in place, so no file is replaced. So maybe we could try to mount it and see what happens...